app/cluster/state/etcd.go

```diff
@@ -267,14 +267,27 @@ func (manager *ETCDManager) unmarshalWorkspace(raw []byte) workspace.ChangeEvent
 			if err != nil {
 				return fmt.Errorf("marshal ack value: %w", err)
 			}
-			manager.logger.Infof("Workspace ID Change Acknowledgement (error: %v) Key: %s", ackErr != nil, req.AckKey)
-			_, err = manager.Client.Put(ctx, req.AckKey, ackValue)
-			if err != nil {
-				manager.logger.Errorf(
-					"Failed to acknowledge workspace ID change (error: %v) for key: %s", ackErr != nil, req.AckKey,
-				)
+			// for backward compatibility, for GATEWAY we acknowledge the request at both
+			// the incoming value contained in ack_key and the new value obtained by appending
+			// the instance ID to the ack_key. This enables the scheduler to ensure all the
+			// replicas of the gateway have acknowledged the request, when running in HA mode
+			ackKeys := []string{req.AckKey}
```

Comment:
I don't think we need backward compatibility for this. We have to update the scheduler before we update server so we should be good I guess. Can we get rid of this?